test/e2e/apimachinery/apply.go

```diff
@@ -0,0 +1,303 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apimachinery
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	"github.com/onsi/ginkgo"
+
+	// ensure libs have a chance to initialize
+	_ "github.com/stretchr/testify/assert"
+)
+
+var _ = SIGDescribe("ServerSideApply", func() {
+	f := framework.NewDefaultFramework("apply")
+
+	var client clientset.Interface
+	var ns string
+
+	ginkgo.BeforeEach(func() {
+		client = f.ClientSet
+		ns = f.Namespace.Name
+	})
+
+	ginkgo.AfterEach(func() {
+	})
+
+	ginkgo.It("PATCH requests with the apply content type will create the object if it does not already exist", func() {
```

Comment:
We can start getting the test identifiers and conformance names and descriptions that we will need for conformance ready now too if we have time.

Descriptions should use rfc2119 language (MUST, MAY, SHOULD) I believe. More details at https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md#conformance-test-comment-metadata. I don't see much guidance on how to format testname..

For test identifiers `ginkgo.It(` in intended to be interpreted as starting a sentence with "It..", e.g. "It should be able to create an object using an patch request if it doesn't exist already", where "it" typically refers to whatever was passed to `SIGDescribe()`.
 
Some examples: https://github.com/kubernetes/kubernetes/pull/81864/files

Reply:
Thanks for the suggestion and links. I'll try to include the conformance block in this PR as well.